From 13cffb0ce708b4465a14b4ee0fab2277e0dc5417 Mon Sep 17 00:00:00 2001
From: "Wael M. Nasreddine" <wael.nasreddine@gmail.com>
Date: Wed, 20 Feb 2019 23:37:03 -0800
Subject: [PATCH] nix-darwin: login as the user when activating

---
 nix-darwin/default.nix | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nix-darwin/default.nix b/nix-darwin/default.nix
index 810b3e5e..98d41a3f 100644
--- a/nix-darwin/default.nix
+++ b/nix-darwin/default.nix
@@ -33,7 +33,7 @@ in
     system.activationScripts.extraActivation.text =
       lib.concatStringsSep "\n" (lib.mapAttrsToList (username: usercfg: ''
         echo Activating home-manager configuration for ${username}
-        sudo -u ${username} ${usercfg.home.activationPackage}/activate
+        sudo -u ${username} -i ${usercfg.home.activationPackage}/activate
       '') cfg.users);
   };
 }
