diff --git a/src/init.cpp b/src/init.cpp
index 5114be8..63148b1 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -182,7 +182,8 @@ void initializeDataAndRules (
   // Load the configuration data.
   File configFile (dbLocation);
   configFile += "timewarrior.cfg";
-  configFile.create (0600);
+  if (! configFile.exists())
+    configFile.create (0600);
   rules.load (configFile._data);
 
   // This value is not written out to disk, as there would be no point. Having
