diff --git a/pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix b/pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix
index 04340ac4e89..8301dcd49b8 100644
--- a/pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix
+++ b/pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix
@@ -275,15 +275,20 @@ self: super:
     old:
       let
         blas = old.passthru.args.blas or pkgs.openblasCompat;
-        lapack = old.passthru.args.lapack or pkgs.openblasCompat;
+        blasImplementation = lib.nameFromURL blas.name "-";
         cfg = pkgs.writeTextFile {
           name = "site.cfg";
-          text = (lib.generators.toINI {} {
-            ${blas.implementation} = {
-              include_dirs = "${blas}/include:${lapack}/include";
-              library_dirs = "${blas}/lib:${lapack}/lib";
-            };
-          });
+          text = (
+            lib.generators.toINI {} {
+              ${blasImplementation} = {
+                include_dirs = "${blas}/include";
+                library_dirs = "${blas}/lib";
+              } // lib.optionalAttrs (blasImplementation == "mkl") {
+                mkl_libs = "mkl_rt";
+                lapack_libs = "";
+              };
+            }
+          );
         };
       in
         {
@@ -294,8 +299,8 @@ self: super:
             ln -s ${cfg} site.cfg
           '';
           passthru = old.passthru // {
-            blsaImplementation = blas.implementation;
-            inherit blas cfg;
+            blas = blas;
+            inherit blasImplementation cfg;
           };
         }
   );
