diff --git a/pkgs/tools/package-management/nix/darwin-never-use-tmpdir.patch b/pkgs/tools/package-management/nix/darwin-never-use-tmpdir.patch
new file mode 100644
index 00000000000..7eb77a6150d
--- /dev/null
+++ b/pkgs/tools/package-management/nix/darwin-never-use-tmpdir.patch
@@ -0,0 +1,12 @@
+diff -Naur a/src/libmain/shared.cc b/src/libmain/shared.cc
+--- a/src/libmain/shared.cc	2020-04-12 13:44:54.000000000 -0700
++++ b/src/libmain/shared.cc	2020-04-12 14:01:57.000000000 -0700
+@@ -155,7 +155,7 @@
+        sshd). This breaks build users because they don't have access
+        to the TMPDIR, in particular in ‘nix-store --serve’. */
+ #if __APPLE__
+-    if (getuid() == 0 && hasPrefix(getEnv("TMPDIR"), "/var/folders/"))
++    if (hasPrefix(getEnv("TMPDIR", "/tmp"), "/var/folders/"))
+         unsetenv("TMPDIR");
+ #endif
+ }
diff --git a/pkgs/tools/package-management/nix/default.nix b/pkgs/tools/package-management/nix/default.nix
index 96fd1bebdd4..b9d68cc0c43 100644
--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -22,12 +22,13 @@ common =
   , withAWS ? stdenv.isLinux || stdenv.isDarwin, aws-sdk-cpp
 
   , name, suffix ? "", src, crates ? null
+  , patches ? []
 
   }:
   let
      sh = busybox-sandbox-shell;
      nix = stdenv.mkDerivation rec {
-      inherit name src;
+      inherit name src patches;
       version = lib.getVersion name;
 
       is24 = lib.versionAtLeast version "2.4pre";
@@ -181,6 +182,19 @@ in rec {
       sha256 = "1c626a0de0acc69830b1891ec4d3c96aabe673b2a9fd04cef84f2304d05ad00d";
     };
 
+    # On Darwin, do not use /var/folders because the resulting path is longer
+    # than allowed for a unix socket.
+    # https://github.com/NixOS/nix/pull/3488.patch
+    patches = lib.optionals stdenv.hostPlatform.isDarwin [
+      # a combination of
+      # https://github.com/LnL7/nix/commit/abb8ef619ba2fab3ae16fb5b5430215905bac723
+      # and
+      # https://github.com/LnL7/nix/commit/4d9db420ffc9bd48da107a61c093b0d65d9d8db1
+      # adapted to work without
+      # https://github.com/NixOS/nix/commit/ba87b08f8529e4d9f8c58d8c625152058ceadb75
+      ./darwin-never-use-tmpdir.patch
+    ];
+
     inherit storeDir stateDir confDir boehmgc;
   } // stdenv.lib.optionalAttrs stdenv.cc.isClang {
     stdenv = llvmPackages_6.stdenv;
