diff --git a/nix-darwin/default.nix b/nix-darwin/default.nix
index 810b3e5..695cef6 100644
--- a/nix-darwin/default.nix
+++ b/nix-darwin/default.nix
@@ -11,6 +11,8 @@ let
 
     config = {
       submoduleSupport.enable = true;
+      submoduleSupport.externalPackageInstall = true;
+
       home.username = config.users.users.${name}.name;
       home.homeDirectory = config.users.users.${name}.home;
     };
