From d0a11e90e769516da3c2bdb928515ec5d9345820 Mon Sep 17 00:00:00 2001
From: "Wael M. Nasreddine" <wael.nasreddine@gmail.com>
Date: Tue, 19 Feb 2019 10:05:47 -0800
Subject: [PATCH] nix-darwin: activate home-manager through postActivation

---
 nix-darwin/default.nix | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nix-darwin/default.nix b/nix-darwin/default.nix
index 810b3e5e..b637d34b 100644
--- a/nix-darwin/default.nix
+++ b/nix-darwin/default.nix
@@ -30,8 +30,8 @@ in
   };
 
   config = mkIf (cfg.users != {}) {
-    system.activationScripts.extraActivation.text =
-      lib.concatStringsSep "\n" (lib.mapAttrsToList (username: usercfg: ''
+    system.activationScripts.postActivation.text =
+      concatStringsSep "\n" (mapAttrsToList (username: usercfg: ''
         echo Activating home-manager configuration for ${username}
         sudo -u ${username} ${usercfg.home.activationPackage}/activate
       '') cfg.users);
