diff --git a/pkgs/development/python-modules/jsbeautifier/default.nix b/pkgs/development/python-modules/jsbeautifier/default.nix
index 9b2d3a5b832..b73d24e3fb6 100644
--- a/pkgs/development/python-modules/jsbeautifier/default.nix
+++ b/pkgs/development/python-modules/jsbeautifier/default.nix
@@ -13,6 +13,8 @@ buildPythonApplication rec {
     sha256 = "7d02baa9b0459bf9c5407c1b99ad5566de04a3b664b18a58ac64f52832034204";
   };
 
+  doCheck = false;
+
   meta = with lib; {
     homepage    = "http://jsbeautifier.org";
     description = "JavaScript unobfuscator and beautifier.";
